.github/workflows/build.yaml

```diff
@@ -77,3 +76,51 @@ jobs:
           labels: ${{ steps.meta.outputs.labels }}
           build-args: |
             SLIM_IMAGE=${{ steps.meta.outputs.tags }}-slim
+
+      - name: Fetch all tags and store them
+        run: |
+          # List all tags
+          tags=$(git tag -l)
+          echo "All tags:"
+          echo "$tags"
+
+          # Save tags in an environment variable for later use
+          echo "ALL_TAGS=$(echo $tags | tr '\n' ' ')" >> $GITHUB_ENV
+
+      - name: Retain last two minor versions
+        run: |
+          # Fetch all tags in the format X.Y.Z
+          echo "Using stored tags..."
+          all_tags="${{ env.ALL_TAGS }}"
+          echo "All tags:"
+          echo "$all_tags"
+
+          # Extract the minor versions (X.Y) from tags in the format vX.X.X
+          latest_two_minors=$(echo "$all_tags" | tr ' ' '\n' | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+$' | sed -E 's/^v([0-9]+\.[0-9]+)\.[0-9]+$/\1/' | uniq | tail -n 2)
+          echo "Last two minor versions:"
+          echo "$latest_two_minors"
+
+          keep_tags=""
+          for minor in $latest_two_minors; do
+            echo "Processing minor version: $minor"
+            latest_patch=$(echo "$all_tags" | tr ' ' '\n' | grep "^v$minor\." | sort -V | tail -n 1)
```

Comment:
I think we can probably just do wildcard on all patches in the last two minors

Reply:
I'm not sure what you mean

Reply:
If I do a wildcard on the last two minors, I'd get all of the patches for those minors. Is that what we want?

Reply:
Yes, someone might be expecting a slightly older patch than whatever is latest there

Reply:
ah, I asked earlier, but inferred incorrectly, I just didn't know if we were trying to reduce space or not. I'll add a wildcard

Reply:
last two minors reduces space quite a bit. I think it's a decent policy, I'm not particularly concerned about space yet but wanted to limit the amount of uploads you had to do from dockerhub